### PR TITLE
[tiff] use vcpkg wrapper for all configs, remove dead code

### PR DESCRIPTION
**Pull request by cenit**
since `tiff` already has a `vcpkg-cmake-wrapper.cmake`, the special code inside `scripts/buildsystems/vcpkg.cmake` never runs. But the code inside `vcpkg-cmake-wrapper.cmake` is wrapped for static installs, while it is always necessary. This PR fixes this confusion.
Also, why not move all the special cases for particular ports, described inside `scripts/buildsystems/vcpkg.cmake`, into different `vcpkg-cmake-wrapper.cmake` for each involved port?

### CODE REVIEW

**Comment by Rastaban** [comment]
Resolved merge conflicts so the CI tests could run.

File: ports/tiff/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-        endif()
-    endif()
-endif()
+_find_package(${ARGS})
+
+find_package(LibLZMA)
+find_package(JPEG)
+find_package(ZLIB)
+if(TARGET TIFF::TIFF)
+  set_property(TARGET TIFF::TIFF APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${LIBLZMA_LIBRARIES} JPEG::JPEG ZLIB::ZLIB)
```
Comment (Neumann-A) [inline ports/tiff/vcpkg-cmake-wrapper.cmake]: Question: Is this a cmake bug in findTIFF? (missing the dependencies)
Maybe report that in the cmake bug tracker?

**Comment by cenit** [comment]
I have to investigate it more. I just found this problem when bisecting the latest message in the opencv4 thread. In the meantime I thought that removing dead code would have been good, just to not pretend it was there to fix anything...

**Comment by cenit** [comment]
In case it is a bug, I will submit a bug and a MR (their PR) for the problem for sure

**Comment by vicroms** [comment]
I see regressions on `osg-qt` and `vxl` for dynamic 32-, and 64-bit Windows .

`osg-qt` regressions are because of the `qt5-base` issue we had last week that has been now mitigated.

And this is the message for `vxl`:

```
Building package vxl[core]:x86-windows... 
Could not locate cached archive: C:\vsts\_work\2\s\archives\da\daa17c9cb226687f4a284819da508b7c94e56d94.zip
CMake Error at ports/vxl/portfile.cmake:4 (message):
  Can't build VXL with non built-in OpenJpeg in current version.  Please
  remove OpenJpeg, and try install VXL again.
Call Stack (most recent call first):
  scripts/ports.cmake:71 (include)
```

**Comment by PhoebeHui** [comment]
@cenit, thanks for contributing to vcpkg!

Could you also update CONTROL version in https://github.com/Microsoft/vcpkg/blob/master/ports/tiff/CONTROL?

**Comment by Rastaban** [comment]
The osg-qt issue are cleared up, but I still see vxl failures in latest commit.

**Comment by vicroms** [comment]
@Rastaban  that is by design

https://github.com/Microsoft/vcpkg/blob/fc9aa1528734ba86758708738c3839c1066ea861/ports/vxl/portfile.cmake#L3-L5

**Comment by Rastaban** [comment]
Ah, I see.  In that case this PR has no build regressions in the CI system.

**Pull request merged**